Library/Homebrew/formula.rb

```diff
@@ -2429,10 +2430,45 @@ def to_hash_with_variations
       end
     end
 
-    hash["variations"] = variations
+    hash["variations"] = @variations_hash
     hash
   end
 
+  # Builds internal json v3 hash.
+  #
+  # @private
+  def to_api_hash
+    api_hash = to_hash_with_variations.except(
+      # Included in the top-level of the `homebrew-core.json` file
+      # created by the `brew generate-formula-api` command.
+      "aliases",
+      "oldname",
+      "oldnames",
+      "tap",
+      "tap_git_head",
+    )
+
+    # Remove keys that default to zero.
+    %w[revision version_scheme].each do |key|
+      next unless api_hash[key].zero?
+
+      api_hash.delete(key)
+    end
+
+    if (stable_bottle = api_hash.dig("bottle", "stable"))
+      stable_bottle["files"] = stable_bottle["files"].transform_values do |file|
+        # The "url" is not used here and takes up a lot of space.
```

Comment:
Could also probably remove the `root_url` given it's the same for every bottle in the tap.

Reply:
Yeah, that could probably just be at the top-level.

Reply:
Actually, it's not used at all.

https://github.com/Homebrew/brew/blob/86e1c8aacf6196e10cef705fe01fc8d8f6737ec4/Library/Homebrew/formulary.rb#L293-L297

Reply:
Yeah it was removed for security reasons prior to the introduction of JWS. Those reasons aren't really relevant anymore but equally the API should be able to behave like `brew bottle` and omit `root_url` if it is the default.

Even that code that's there now on the Formulary feels unnecessary - surely we can omit it like we omit it for Ruby formulae.